app/controllers/DeclarationController.scala

```diff
@@ -47,7 +48,7 @@ class DeclarationController @Inject()(
     Future.successful(Ok(view()))
   }
 
-  def onSubmit: Action[AnyContent] = (identify andThen getData andThen requireData).async { implicit request =>
+  def onSubmit: Action[AnyContent] = (identify andThen getData andThen requireDataSubmit).async { implicit request =>
```

Comment:
i guess RegistrationDataRequiredAction not required here, could you please check it